Handle unknown version number and simplify version check

zzamboni opened the pull request:
- current-commit now returns `unknown` if no version is detected.
- check-commit now handles `unknown` version and skips the check.
- Simplify parsing in check-commit by not requesting the headers.

zzamboni committed:
Handle unknown version number and simplify version check

- current-commit now returns `unknown` if no version is detected.
- check-commit now handles `unknown` version and skips the check.
- Simplify parsing in check-commit by not requesting the headers.

iwoloschin commented on update.elv:
```diff
@@ -32,26 +32,25 @@ update-message = 'Elvish Upgrade Available - update:build-HEAD'
+    echo (styled "Your elvish does not report a version number in elvish -buildinfo" red)
+  } else {
+    error = ?(
+      json = (curl -s https://api.github.com/repos/elves/elvish/compare/$commit...master | from-json)
```
I'd like to keep the headers, at least for now, as we can use it to make conditional requests that might help avoid hitting the rate limit.  This is obviously not implemented yet, but I think it's probably next on my list.

Conditional Requests: https://developer.github.com/v3/#conditional-requests

zzamboni replied:
I took them out because the split-on-empty-line was failing for me when running from work (behind a proxy). In this case, the output of `curl -s -i` starts like this:
```
HTTP/1.0 200 Connection established

HTTP/1.1 200 OK
Server: GitHub.com
Date: Mon, 04 Jun 2018 17:51:31 GMT
Content-Type: application/json; charset=utf-8
Content-Length: 8620
Status: 200 OK
X-RateLimit-Limit: 60
...
```

The first line (separated by a blank line) is the response from the proxy server, followed by the real headers. This means that the indexing was failing. I guess this can be fixed by using `$compare[-1]` instead of `$compare[1]`. I'll propose a fix.

zzamboni committed:
Restored headers

Restored option -i of curl to get the headers, but fixed the access to
the message parts to skip any extraneous header blocks at the
beginning (e.g. from a proxy).
